index/upsidedown/row.go

```diff
@@ -315,35 +315,16 @@ func NewDictionaryRowK(key []byte) (*DictionaryRow, error) {
 }
 
 func (dr *DictionaryRow) parseDictionaryK(key []byte) error {
-	buf := bytes.NewBuffer(key)
-	_, err := buf.ReadByte() // type
-	if err != nil {
-		return err
-	}
-
-	err = binary.Read(buf, binary.LittleEndian, &dr.field)
-	if err != nil {
-		return err
+	dr.field = binary.LittleEndian.Uint16(key[1:3])
+	if dr.term != nil {
+		dr.term = dr.term[:0]
 	}
-
-	dr.term, err = buf.ReadBytes(ByteSeparator)
-	// there is no separator expected here, should get EOF
-	if err != io.EOF {
-		return err
-	}
-
+	dr.term = append(dr.term, key[3:]...)
 	return nil
 }
 
 func (dr *DictionaryRow) parseDictionaryV(value []byte) error {
-	buf := bytes.NewBuffer(value)
-
-	count, err := binary.ReadUvarint(buf)
-	if err != nil {
-		return err
-	}
-	dr.count = count
-
+	dr.count, _ = binary.Uvarint(value)
```

Comment:
Oh I assumed it was an error since the corresponding read version returned an error.  Maybe there is no error case I'm not sure.